testing/logging/test_reporting.py

```diff
@@ -410,3 +413,55 @@ def test_log_file(request):
         contents = rfh.read()
         assert "This log message will be shown" in contents
         assert "This log message won't be shown" not in contents
+
+
+@pytest.mark.parametrize('has_capture_manager', [True, False])
+def test_live_logging_suspends_capture(has_capture_manager, request):
+    """Test that capture manager is suspended when we emitting messages for live logging.
+
+    This tests the implementation calls instead of behavior because it is difficult/impossible to do it using
+    ``testdir`` facilities because they do their own capturing.
+
+    We parametrize the test to also make sure _LiveLoggingStreamHandler works correctly if no capture manager plugin
+    is installed.
+    """
+    import logging
+    from functools import partial
+    from _pytest.capture import CaptureManager
+    from _pytest.logging import _LiveLoggingStreamHandler
+
+    if six.PY2:
+        # need to use the 'generic' StringIO instead of io.StringIO because we might receive both bytes
+        # and unicode objects; io.StringIO only accepts unicode
+        from StringIO import StringIO
+    else:
+        from io import StringIO
+
+    class MockCaptureManager:
+        calls = []
+
+        def suspend_global_capture(self):
+            self.calls.append('suspend_global_capture')
+
+        def resume_global_capture(self):
+            self.calls.append('resume_global_capture')
+
+    # sanity check
+    assert CaptureManager.suspend_capture_item
+    assert CaptureManager.resume_global_capture
+
+    capture_manager = MockCaptureManager() if has_capture_manager else None
+    out_file = StringIO()
+
+    handler = _LiveLoggingStreamHandler(out_file, capture_manager)
+
+    logger = logging.getLogger(__file__ + '.test_live_logging_suspends_capture')
```

Comment:
you probably wanted to write `__name__` instead of `__file__`, right?

Reply:
Definitely, thanks